File: old/c++/MatrixMultiplication.cpp
#include <iostream>
#include <vector>

std::vector<std::vector<int>> multiplyMatrix(std::vector<std::vector<int>> matrixA, std::vector<std::vector<int>> matrixB) {

    std::vector<std::vector<int>> matrixC(matrixA.size());

    for (int r = 0; r < matrixA.size(); r++) {
        matrixC[r].resize(matrixB[0].size());
    }

    for (int y = 0; y < matrixA.size(); y++) {
        for (int x = 0; x < matrixA[0].size(); x++) {
            for (int n = 0; n < matrixB.size(); n++) {

                matrixC[y][x] += matrixA[y][n] * matrixB[n][x];

            }
        }
    }

    return matrixC;
}

void printMatrix(std::vector<std::vector<int>> matrix){
    for (int y = 0; y < matrix.size(); y++) {
        for (int x = 0; x < matrix[0].size(); x++) {

            std::cout << matrix[y][x] << " ";
            
        }
        std::cout << "" << std::endl;
    }
}

int main(int argc, char const *argv[]) {
    
    std::vector<std::vector<int>> matrixA = {{5, 6, 9},
                                             {7, 8, 9}};

    std::vector<std::vector<int>> matrixB = {{10, 15},
                                             {22, 28},
                                             {15, 15}};

    std::vector<std::vector<int>> result = multiplyMatrix(matrixA, matrixB);


    printMatrix(matrixA);
    std::cout << "*" << std::endl;
    printMatrix(matrixB);
    std::cout << "=" << std::endl;
    printMatrix(result);

    return 0; 
}

File: old/java/MatrixMultiplication.java

public class MatrixMultiplication {


    public static void main(String[] args) {
        
        // Matrix definitions
        int[][] matrixA = {{5, 6, 9},
                           {7, 8, 9}};

        int[][] matrixB = {{10, 15},
                           {22, 28},
                           {15, 15}};

        int[][] matrixC = multiplyMatrix(matrixA, matrixB);


        // Terminal output
        printMatrix(matrixA);
        System.out.println("*");
        printMatrix(matrixB);
        System.out.println("=");
        printMatrix(matrixC);

    }

    public static void printMatrix(int[][] matrix) {
        for (int y = 0; y < matrix.length; y++) {
            for (int x = 0; x < matrix[0].length; x++) {
    
                System.out.print(matrix[y][x] + " ");
                
            }
            System.out.print("\n");
        }
    }


    public static int[][] multiplyMatrix(int[][] matrixA, int[][] matrixB) {

        // Initialize matrixC
        int[][] matrixC = new int[matrixB[0].length][matrixA.length];

        // Multiply matrixA with matrixB
        for (int y = 0; y < matrixA.length; y++) {
            for (int x = 0; x < matrixB[0].length; x++) {
                for (int n = 0; n < matrixB.length; n++) {
    
                    matrixC[y][x] += matrixA[y][n] * matrixB[n][x];
    
                }
            }
        }

        return matrixC;
    }

}